### PR TITLE
 Initial Setup

### PR DESCRIPTION
**Pull request by codenirvana**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm
+
+*.iml
+
+## Directory-based project format:
+.idea/
+# if you remove the above rule, at least ignore the following:
+
+# User-specific stuff:
+# .idea/workspace.xml
```
Comment (kunagpal) [inline .gitignore:11]: This `.idea` section can be removed, as it is already covered on line 7.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,157 @@
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm
+
+*.iml
+
+## Directory-based project format:
+.idea/
```
Comment (kunagpal) [inline .gitignore:7]: Also add `.vscode` if not done already.

**Review by kunagpal — CHANGES_REQUESTED** [review]

**Review by kunagpal — APPROVED** [review]

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# editorconfig.org
+root = true
+
+[*.json]
+indent_style = space
+indent_size = 4
```
Comment (shamasis) [inline .editorconfig:6]: We could now have 2 spaces for json files.

Comment (kunagpal) [reply .editorconfig:6]: @shamasis Yes, all such rectifications will occur in subsequent pull requests.

**Pull request merged**